### PR TITLE
Two credit card amount error and recurrence payment

### PR DESCRIPTION
**Pull request by michelpl**
# What?

Fix two credit card total amount and fix recurrence payment errors

# Why?

After last update, two credit cards amount was not correct and recurrence payment was not working.

# How?
Dealing with more payment flows and cleaning code.

### CODE REVIEW

File: app/code/community/Uecommerce/Mundipagg/Model/Api.php [hunk 1]
```diff
@@ -1,5 +1,32 @@
 <?php
-
+/**
+ * Uecommerce
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Api.php]: Uecommerce?

File: app/code/community/Uecommerce/Mundipagg/Model/Api.php [hunk 2]
```diff
@@ -103,7 +121,7 @@ public function creditCardTransaction($order, $data, Uecommerce_Mundipagg_Model_
 					$creditcardTransactionData->CreditCard->SecurityCode = $paymentData['SecurityCode']; // Código de segurança
 					$creditcardTransactionData->CreditCard->ExpMonth = $paymentData['ExpMonth']; // Mês Exp
 					$creditcardTransactionData->CreditCard->ExpYear = $paymentData['ExpYear']; // Ano Exp 
-					$creditcardTransactionData->CreditCard->CreditCardBrand = $this->creditCardBrands[$paymentData['CreditCardBrandEnum']]; // Bandeira
+					$creditcardTransactionData->CreditCard->CreditCardBrand = $paymentData['CreditCardBrandEnum']; // Bandeira do cartão : Visa ,MasterCard ,Hipercard ,Amex */
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Api.php]: Start comment with // and end with */ ??

File: app/code/community/Uecommerce/Mundipagg/Model/Observer.php [hunk 3]
```diff
@@ -141,7 +141,7 @@ public function removePaymentMethods($observer) {
 	}
 
 	public function removeInterest($observer) {
-		$session = Mage::getSingleton('admin/session');
+		/*$session = Mage::getSingleton('admin/session');
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Observer.php]: Please, remove commented block code. If you need to use this code, get in history of commit.

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php [hunk 4]
```diff
@@ -1,1101 +1,1063 @@
...
+    /**
+     * Order payment abstract method
+     *
+     * @param Varien_Object $payment
+     * @param float         $amount
+     *
+     * @return Mage_Payment_Model_Abstract
+     */
+    public function order(Varien_Object $payment, $amount) {
+        if (!$this->canOrder()) {
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Is possible reduce this method.

For example, all cases of switch can turn a specific methods.

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php [hunk 5]
```diff
@@ -1,1101 +1,1063 @@
...
+    }
+
+    /**
+     * Online capture payment abstract methodl
+     *
+     * @param Varien_Object $payment
+     * @return $this
+     */
+    public function captureOnline(Varien_Object $payment) {
+        /* @var Uecommerce_Mundipagg_Helper_Data $helper */
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Create a protected method to get helper with dockbloc don't is a good idea to resolve this problem?

You used this docblock in others place for the same purpose.

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php [hunk 6]
```diff
@@ -1,1101 +1,1063 @@
...
+        if ($this->getAntiFraud() == 1) {
+            Mage::throwException($helper->__('You cannot capture having anti fraud activated.'));
+        }
+
+        // Already captured
+        if ($payment->getAdditionalInformation('CreditCardTransactionStatusEnum') == 'Captured' || $payment->getAdditionalInformation('CreditCardTransactionStatus') == 'Captured') {
+            return $this;
+        }
+
+        /* @var Mage_Sales_Model_Order_Payment $payment */
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Don't is possible put this docblock in method getAdditonalInformation?

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php [hunk 7]
```diff
@@ -1,1101 +1,1063 @@
...
+        }
+
+        $captureNotAllowedMsg = $helper->__('Capture was not authorized in MundiPagg');
+        $txnsNotAuthorized = 0;
+
+        foreach ($orderkeys as $orderkey) {
+            $data['OrderKey'] = $orderkey;
+
+            //Call Gateway Api
+            /* @var Uecommerce_Mundipagg_Model_Api $api */
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Create a protected method to get api with dockbloc don't is a good idea to resolve this problem?

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php [hunk 8]
```diff
@@ -1,1101 +1,1063 @@
...
+        } elseif ($txnsNotAuthorized > 1) {
+            Mage::throwException($helper->__('Capture partial authorized'));
+        }
+
+        $this->closeAuthorizationTxns($payment->getOrder());
+
+        // if has just 1 invoice, update his grand total, adding the credit cards interests
+        if (count($payment->getOrder()->getInvoiceCollection()) === 1) {
+
+            /* @var Mage_Sales_Model_Order_Invoice $invoice */
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Don't is possible put this docblock in method getItems? This is a internal method of Magento?

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php [hunk 9]
```diff
@@ -1,1101 +1,1063 @@
...
+            $invoice = $payment->getOrder()->getInvoiceCollection()->getItems()[0];
+            $this->equalizeInvoiceTotals($invoice);
+        }
+    }
+
+    public function closeAuthorizationTxns(Mage_Sales_Model_Order $order) {
+        $txnsCollection = Mage::getModel('sales/order_payment_transaction')
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Create a protected method to get this model, with dockbloc in method, don't is a good idea to resolve this problem?

i.e.: `@return order_payment_transaction[]`

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php [hunk 10]
```diff
@@ -1,1101 +1,1063 @@
...
+
+        $helper = Mage::helper('mundipagg');
+        $result = $helper->issetOr($resultPayment['result'], false);
+
+        if ($result === false) {
+            return $this->integrationTimeOut($order, $payment);
+        }
+
+        // Return error
+        if (isset($resultPayment['error']
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Don't is possible break this method in another methods?

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php [hunk 11]
```diff
@@ -1200,1354 +1157,1325 @@ public function doPayment($payment, $order) {
...
-		if (!isset($info['IntegrationError'])) {
-			return;
-		}
-
-		try {
-			$info = $order->getPayment()->getAdditionalInformation();
+                    break;
+
+                // 1CreditCards
+                case 1: // AuthAndCapture
```
Comment (vitormattos) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Another good case to split in multiple methods

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php (hunk 5 above)
Comment (michelpl) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Não fui eu que adicionei esse código. Veio quando dei merge.

File: app/code/community/Uecommerce/Mundipagg/Model/Standard.php (hunk 4 above)
Comment (michelpl) [inline app/code/community/Uecommerce/Mundipagg/Model/Standard.php]: Legal

**Pull request merged**